src/Tmds.Ssh/SftpClient.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+using System.Buffers;
+using System.Buffers.Binary;
+
+namespace Tmds.Ssh
+{
+    abstract class SftpOperation
+    {
+        public abstract ValueTask HandleResponse(SftpPacketType type, ReadOnlySequence<byte> fields);
+
+        protected static Exception CreateExceptionForStatus(ReadOnlySequence<byte> fields)
+        {
+            return new SshException("TODO"); // TODO: use fields
```

Comment:
The TODO is about adding a `ParseStatusFields` method that parses  https://tools.ietf.org/html/draft-ietf-secsh-filexfer-13#section-9.1.
Let's read
```
       uint32 error/status code
       string error message (ISO-10646 UTF-8 [RFC-2279])
```
We can add an enum for the status code: e.g. `SftpErrorCode` with the error codes from that section.
The `ErrorCode` can be added as a property on the `SshException`.
We can use error message as the `Message` for the `SshException`.

Reply:
Okey, so for a status success do nothing and any error results in an exception

Reply:
Also you are linking a draft to version 6, but I think we're doing version 3 now?

Reply:
But what if we don't want to throw exception on every STATUS message? maybe parse fields first and if its not success then pass it to the CreateExceptionForStatus?

Reply:
Yes, that makes sense. Now we have such a case with SftpFile.CloseAsync.